File: services/payment/src/main/java/com/example/iter/payment/dto/response/PaymentConfirmResponse.java
package com.example.iter.payment.dto.response;

import com.example.iter.payment.domain.entity.Payment;
import com.example.iter.payment.api.PaymentStatus;
import com.example.iter.reservation.api.RentalInfo;
import com.example.iter.reservation.api.RentalStatus;

import java.math.BigDecimal;
import java.time.LocalDateTime;

public record PaymentConfirmResponse(
        Long rentalId,
        Long paymentId,
        String paymentKey,
        BigDecimal amount,
        PaymentStatus paymentStatus,
        LocalDateTime paidAt,
        RentalStatus rentalStatus
) {
    public static PaymentConfirmResponse of( RentalInfo rental, Payment payment ) {
        return new PaymentConfirmResponse(
                rental.rentalId(),
                payment.getId(),
                payment.getPaymentKey(),
                payment.getAmount(),
                payment.getStatus(),
                payment.getPaidAt(),
                rental.status()
        );
    }
}

File: services/payment/src/main/kotlin/com/example/iter/payment/dto/response/PaymentConfirmResponse.kt
package com.example.iter.payment.dto.response

import com.example.iter.payment.api.PaymentStatus
import com.example.iter.payment.domain.entity.Payment
import com.example.iter.reservation.api.RentalInfo
import com.example.iter.reservation.api.RentalStatus
import java.math.BigDecimal
import java.time.LocalDateTime

@JvmRecord
data class PaymentConfirmResponse(
    val rentalId: Long,
    val paymentId: Long?,
    val paymentKey: String?,
    val amount: BigDecimal,
    val paymentStatus: PaymentStatus,
    val paidAt: LocalDateTime?,
    val rentalStatus: RentalStatus,
) {
    companion object {
        @JvmStatic
        fun of(rental: RentalInfo, payment: Payment): PaymentConfirmResponse = PaymentConfirmResponse(
            rental.rentalId,
            payment.id,
            payment.paymentKey,
            payment.amount,
            payment.status,
            payment.paidAt,
            rental.status,
        )
    }
}
